From 8f63da3f62d25ab3c868d806feb5ae83d64a74ca Mon Sep 17 00:00:00 2001
From: Pierre-Jean Texier' via swupdate <swupdate@googlegroups.com>
Date: Thu, 2 May 2019 21:09:40 +0200
Subject: [libubootenv][PATCH v3] fw_printenv: set 'u-boot-initial-env' file as
 default

Since commit [1] in U-Boot, this is the file with the initial environment
delivered with the bootloader.
So, let's set this name as default.

tested with buildroot as follows:

1) # fw_printenv
Cannot read environment, using default
baudrate=115200
...

2) # fw_setenv foo bar
Cannot read environment, using default

3) # fw_printenv foo
foo=bar

[1] https://github.com/u-boot/u-boot/commit/bdaa73a5b3923257add182b4ab8058dbfa33421b#diff-b67911656ef5d18c4ae36cb6741b7965

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
---
Changes v1 -> v2:
	- fix commit log
	
Changes v2 -> v3:
	- fix commit log :(
	
 src/fw_printenv.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/fw_printenv.c b/src/fw_printenv.c
index 361d150..5f75db7 100644
--- a/src/fw_printenv.c
+++ b/src/fw_printenv.c
@@ -121,6 +121,9 @@ int main (int argc, char **argv) {
 		exit (ret);
 	}
 
+	if (!defenvfile)
+		defenvfile = "/etc/u-boot-initial-env";
+
 	if ((ret = libuboot_open(ctx)) < 0) {
 		fprintf(stderr, "Cannot read environment, using default\n");
 		if ((ret = libuboot_load_file(ctx, defenvfile)) < 0) {
-- 
2.7.4

